tests/unit/plugins/kubernetes/kubernetes_pod_executor_test.py

```diff
@@ -136,13 +144,18 @@ def test_run(k8s_executor):
                 host_path=V1HostPathVolumeSource(path="/a"),
                 name="host--slash-a",
             )],
+            node_selector={"hello": "world"},
+            affinity=V1Affinity(node_affinity=mock_get_node_affinity.return_value),
         ),
     )
 
     assert k8s_executor.run(task_config) == task_config.pod_name
     assert k8s_executor.kube_client.core.create_namespaced_pod.call_args_list == [
         mock.call(body=expected_pod, namespace='task_processing_tests')
     ]
+    assert mock_get_node_affinity.call_args_list == [
+        mock.call(pvector([dict(key="a_label", operator="In", value=[])])),
+    ]
```

Comment:
thoughts on just relying on the check that create_namespaced_pod was called with the expected body and not checking how get_node_affinity was called?

Reply:
i generally prefer checking how things are called to validate correct args, in addition to checking output. open to changing

Reply:
sounds good! I normally only do one or the other since if you can assert the output you know everything is working as intended and you don't have extra test failures if you happened to change the function signature of something

i'm fine with keeping this as is tho!